frontend/fingertips-frontend/lib/search/searchService.ts

```diff
@@ -0,0 +1,62 @@
+import { IndicatorSearch, IndicatorSearchResult } from './searchResultData';
+import { AzureKeyCredential, SearchClient } from '@azure/search-documents';
+
+type Indicator = {
+  IID: string;
+  Descriptive: {
+    Name: string;
+    Definition: string;
+    DataSource: string;
+  };
+  DataChange: {
+    LastUploadedAt: string;
+  };
+};
+
+const getEnvironmentVariable = (name: string): string => {
+  return (
+    process.env[name] ??
+    (() => {
+      throw new Error(`Missing environment variable ${name}`);
+    })()
+  );
+};
+
+export class SearchService implements IndicatorSearch {
+  readonly searchClient: SearchClient<Indicator>;
+
+  constructor() {
+    const serviceName = getEnvironmentVariable('DHSC_AI_SEARCH_SERVICE_NAME');
+    const indexName = getEnvironmentVariable('DHSC_AI_SEARCH_INDEX_NAME');
+    const apiKey = getEnvironmentVariable('DHSC_AI_SEARCH_API_KEY');
+
+    this.searchClient = new SearchClient<Indicator>(
+      `https://${serviceName}.search.windows.net`,
```

Comment:
I appreciate this was how I did it in the PoC, but I think we're probably better taking the full URL to the search service as an environment variable (e.g. `https://<serviceName>.search.windows.net`), as it means we're not baking in knowledge of the AI search URL structure to our code.